.azure-pipelines/steps/run.yml

```diff
@@ -18,6 +23,22 @@ steps:
 - checkout: self
   fetchDepth: 2
 
+# Set the SKIP_JOB environment variable if this job is supposed to only run
+# when submodules are updated and they were not. The following time consuming
+# tasks will be skipped when the environment variable is present.
+- ${{ if parameters.only_on_updated_submodules }}:
+  - bash: |
+      set -e
+      # Submodules pseudo-files inside git have the 160000 permissions, so when
+      # those files are present in the diff a submodule was updated.
+      if git diff HEAD^ | grep "^index .* 160000" >/dev/null 2>&1; then
```

Comment:
This looks like it only tests the last commit of the PR as opposed to all of them for changing submodules?

Reply:
This checks the merge commit, so any submodule change should be detected.

Reply:
Ah, so `HEAD^` will be the parent that's the old master, not the other parent. Makes sense.

I never know which side `HEAD^` picks with merge commits...